File: app/src/main/java/com/silambarasan/transferdata/java/EndActivity.java
package com.silambarasan.transferdata.java;

import android.content.Intent;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.silambarasan.transferdata.R;

/**
 * Created by SILAMBARASAN on 3/10/2018.
 */

public class EndActivity extends AppCompatActivity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
        super.onCreate(savedInstanceState, persistentState);
        setContentView(R.layout.end_activity_main);

        TextView txtName = (TextView) findViewById(R.id.name);
        TextView txtPin = (TextView) findViewById(R.id.pin);
        TextView txtEmail = (TextView) findViewById(R.id.email);

        Intent i = getIntent();
        // Get all the Data from start activity
        String name = i.getStringExtra("NAME");
        String email = i.getStringExtra("EMAIL");
        String pin = i.getStringExtra("PIN");

        // Displaying Received data
        txtName.setText(name);
        txtPin.setText(pin);
        txtEmail.setText(email);

        findViewById(R.id.btnBack).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }
}

File: app/src/main/java/com/silambarasan/transferdata/kotlin/EndActivity.kt
package com.silambarasan.transferdata.kotlin

import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView
import com.silambarasan.transferdata.R

/**
 * Created by SILAMBARASAN on 3/10/2018.
 */

class EndActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        setContentView(R.layout.end_activity_main)

        val txtName = findViewById<View>(R.id.name) as TextView
        val txtPin = findViewById<View>(R.id.pin) as TextView
        val txtEmail = findViewById<View>(R.id.email) as TextView

        // Get all the Data from start activity
        val name = intent.getStringExtra("NAME")
        val email = intent.getStringExtra("EMAIL")
        val pin = intent.getStringExtra("PIN")

        // Displaying Received data
        txtName.text = name
        txtPin.text = pin
        txtEmail.text = email

        findViewById<View>(R.id.btnBack).setOnClickListener { finish() }
    }
}
